crates/cext/src/transpiler/passes/elide_permutations.rs

```diff
@@ -0,0 +1,195 @@
+// (C) Copyright IBM 2025
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE.txt file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+use crate::pointers::const_ptr_as_ref;
+
+use qiskit_circuit::circuit_data::CircuitData;
+use qiskit_circuit::converters::dag_to_circuit;
+use qiskit_circuit::dag_circuit::DAGCircuit;
+use qiskit_transpiler::passes::run_elide_permutations;
+
+/// The result from ``qk_transpiler_pass_standalone_elide_permutations()``.
+pub struct ElidePermutationsResult {
+    elided: bool,
+    circuit: Option<CircuitData>,
+    permutation: Option<Vec<usize>>,
+}
+
+/// @ingroup QkElidePermutationsResult
+/// Check whether the elide permutations was able to elide anything
+///
+/// @param result a pointer to the result
+///
+/// @returns ``true`` if the ``qk_transpiler_pass_standalone_elide_permutations()`` run elided any gates
+///
+/// # Safety
+///
+/// Behavior is undefined if ``result`` is not a valid, non-null pointer to a
+/// ``QkVF2LayoutResult``.
+#[no_mangle]
+#[cfg(feature = "cbinding")]
+pub unsafe extern "C" fn qk_elide_permutations_result_elided_gates(
+    result: *const ElidePermutationsResult,
+) -> bool {
+    let result = unsafe { const_ptr_as_ref(result) };
+    result.elided
+}
+
+/// @ingroup QkElidePermutationsResult
+/// Get the circuit from the elide permutations result
+///
+/// @param result a pointer to the result of the pass. It must have elided gates as checked by
+/// ``qk_elide_permutations_result_elided_gates()``
+///
+/// @returns A pointer to a circuit with the elide
```

Comment:
A pointer to a circuit with the elide... ?

Maybe we could reword to:

```suggestion
/// @returns A pointer to a circuit with the elision
```